auditable/auditable.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package auditable
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+var (
+	apiPackagePath = "github.com/mattermost/mattermost-server/v6/api4"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "auditable",
+	Doc:  "check if auditable interface is satisfied when passing structs to audit records",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	if pass.Pkg.Path() != apiPackagePath {
+		return nil, nil
+	}
+
+	for _, file := range pass.Files {
+		commentMap := make(map[token.Pos]*ast.CommentGroup)
+		for i := range file.Comments {
+			commentMap[file.Comments[i].Pos()] = file.Comments[i]
+		}
```

Comment:
Minor detail but would it make sense to filter here if the comment matches what we are looking for? Mostly to avoid loading all the comments.